pkg/operator/key.go

```diff
@@ -0,0 +1,136 @@
+package operator
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"fmt"
+	"math/big"
+)
+
+// Curve represents an elliptic curve name that denotes the curve used by
+// the operator keys.
+type Curve int
+
+// Enum that lists all the elliptic curves that can be used with the operator
+// keys.
+const (
+	Undefined Curve = iota
```

Comment:
Do we really need the `Undefined` type of curve? It feels wrong to allow the creation of an operator key without a curve defined.

From what I see `Undefined` type is used only in unit tests. I would feel better removing the unit tests covering `Undefined` checks rather than allowing the operator key to be created without a curve defined because it is not a valid operator key.

Reply:
Here is my reasoning: all types in Go have their zero value (`""` for `string`, `0` for `int`, etc.) so we also need a way to represent the zero value in the `Curve` type once we create an uninitialized `operator.PublicKey`. If we leave just the `Secp256k1` value, this will be the default value of the `Curve` type and `operator.PublicKey`. That feels wrong and this is why I added the `Undefined` value which is the default.

Reply:
I do not think it is wrong to have `Secp256k1` as the default value. A really good practice is to design objects in such a way that if they are initialized and passed around, the code consuming an object can be sure that it's been initialized correctly. Then, such an object can be used without having the consuming code worry about a potential invalid state. Clearly, operator key that has no curve defined *is* in an invalid state. `MarshalUncompressed` does not work, `MarshalCompressed` does not work, and it is not possible to convert such a key to any other key type: network key or chain key because not enough information is provided. We expect the network code and chain code to be able to parse secp256k1 operator key. It feels like a default anyway.

Side note, I was also thinking if the operator key shouldn't be always secp256k1 key but maybe it's too much of a simplification now.

Reply:
Sure, we can remove `Undefined`. Just had a feeling that this will make the operator code more Go-style. However, we must be aware that even if we remove `Undefined` and leave `Secp256k1` to be the zero value, the X and Y coords will still be `nil` by default.

Reply:
See https://github.com/keep-network/keep-core/pull/3025/commits/794310688fd9ebf076e71d3b57a585a44e264e5a